components/gitlab/new-review-request.js

```diff
@@ -0,0 +1,119 @@
+const gitlab = require("https://github.com/PipedreamHQ/pipedream/components/gitlab/gitlab.app.js");
+
+module.exports = {
+  name: "New Review Request (Instant)",
+  description: "Emits an event when a reviewer is added to a merge request",
+  version: "0.0.1",
+  dedupe: "unique",
+  props: {
+    gitlab,
+    projectId: { propDefinition: [gitlab, "projectId"] },
+    http: "$.interface.http",
+    db: "$.service.db",
+  },
+  hooks: {
+    async activate() {
+      const hookParams = {
+        merge_requests_events: true,
+        push_events: false,
+        url: this.http.endpoint,
+      };
+      const opts = {
+        hookParams,
+        projectId: this.projectId,
+      };
+      const { hookId, token } = await this.gitlab.createHook(opts);
+      console.log(
+        `Created "merge request events" webhook for project ID ${this.projectId}.
+        (Hook ID: ${hookId}, endpoint: ${hookParams.url})`
+      );
+      this.db.set("hookId", hookId);
+      this.db.set("token", token);
+    },
+    async deactivate() {
+      const hookId = this.db.get("hookId");
+      const opts = {
+        hookId,
+        projectId: this.projectId,
+      };
+      await this.gitlab.deleteHook(opts);
+      console.log(
+        `Deleted webhook for project ID ${this.projectId}.
+        (Hook ID: ${hookId})`
+      );
+    },
+  },
+  methods: {
+    getNewReviewers(body) {
+      const { action, title } = body.object_attributes;
+
+      // When a merge request is first created, any assignees
+      // in it are interpreted as new review requests.
+      if (action === "open" || action === "reopen") {
+        const { assignees = [] } = body;
+        return assignees;
+      }
+
```

Comment:
Should you return unless action is `updated`? At the very least if it is `closed` we should not emit events?

Reply:
Oh nevermind I see you are looking at `body.changes`